packages/concerto-core/lib/serializer/valuegenerator.js

```diff
@@ -95,6 +96,56 @@ class EmptyValueGenerator {
     getArray(valueSupplier) {
         return [];
     }
+
+    /**
+     * Get a default Regex value.
+     * @param {RegExp} regex A regular expression.
+     * @return {string} a String value.
+     */
+    getRegex(regex) {
+        return new RandExp(regex).gen();
+    }
+
+    /**
+     * Get a default range value.
+     * @param {number} lowerBound the lower bound on the range.
+     * @param {number} upperBound the upper bound on the range.
+     * @param {string} type the number type for the range
+     * @return {number} a String value.
+     */
+    getRange(lowerBound, upperBound, type) {
```

Comment:
Is there another sensible thing to do for the `empty` case? Default to the `lowerBound` perhaps?

Reply:
> I actually like how you did it. Do we check that lowerBound and upperBound are in the proper integer/long range? I don't even know if we check those limits on validation.

Yes that happens in the `randomNumberInRangeWithPrecision` function

I initially tried this, but this causes `NaN` answers when performing precision calculations. For this use case of generating a sample number, I feel that we can restrict the range to lower default extremes, I chose to give Decimals the same range as Longs for convenience